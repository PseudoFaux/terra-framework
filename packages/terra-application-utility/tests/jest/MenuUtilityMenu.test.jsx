import React from 'react';
import MenuUtilityMenu from '../../src/menu/_MenuUtilityMenu';

describe('MenuUtilityMenu', () => {
  const mockOnChange = jest.fn();
  const mockOnDisclose = jest.fn();
  const mockOnRequestClose = jest.fn();
  const config = [{ key: 'key', title: 'title', contentLocation: 'footer', isSelected: false, childKeys: [] }];

  it('should render with default props', () => {
    const wrapper = shallow(
      <MenuUtilityMenu
        initialSelectedKey="key"
        menuItems={config}
        onChange={mockOnChange}
        onDisclose={mockOnDisclose}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with isHeightBounded', () => {
    const wrapper = shallow(
      <MenuUtilityMenu
        initialSelectedKey="key"
        menuItems={config}
        onChange={mockOnChange}
        onDisclose={mockOnDisclose}
        isHeightBounded
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with onRequestClose', () => {
    const wrapper = shallow(
      <MenuUtilityMenu
        initialSelectedKey="key"
        menuItems={config}
        onChange={mockOnChange}
        onDisclose={mockOnDisclose}
        onRequestClose={mockOnRequestClose}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('it should pass in a custom prop', () => {
    const wrapper = shallow(
      <MenuUtilityMenu
        initialSelectedKey="key"
        menuItems={config}
        onChange={mockOnChange}
        onDisclose={mockOnDisclose}
        id="test"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
