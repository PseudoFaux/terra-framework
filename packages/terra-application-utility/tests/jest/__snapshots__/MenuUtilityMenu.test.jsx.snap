// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuUtilityMenu should render a menu utility menu with default props 1`] = `
<ContentContainer
  aria-label="Utility menu"
  className="utility-menu"
  fill={false}
  footer={null}
  header={
    <div
      className="header"
    >
      <span
        className="left-container"
      >
        <span
          className="inner-container"
        >
          <button
            aria-label="Back Button"
            className="back-button"
            onClick={[Function]}
          >
            <IconLeft
              className="icon-left"
              data-name="Layer 1"
              isBidi={true}
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
            />
          </button>
          <span
            className="header-text"
          >
            title
          </span>
        </span>
        <span
          className="right-container"
        >
          <button
            aria-label="Close Button"
            className="close-button"
            onClick={undefined}
          >
            <IconClose
              className="icon-close"
              data-name="Layer 1"
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
            />
          </button>
        </span>
      </span>
      <MenuDivider />
    </div>
  }
  onDisclose={[Function]}
  role="navigation"
/>
`;

exports[`MenuUtilityMenu should render a menu utility menu with default props 2`] = `
<ContentContainer
  aria-label="Utility menu"
  className="utility-menu"
  fill={false}
  footer={null}
  header={
    <div
      className="header"
    >
      <span
        className="left-container"
      >
        <span
          className="inner-container"
        >
          <button
            aria-label="Back Button"
            className="back-button"
            onClick={[Function]}
          >
            <IconLeft
              className="icon-left"
              data-name="Layer 1"
              isBidi={true}
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
            />
          </button>
          <span
            className="header-text"
          >
            title
          </span>
        </span>
        <span
          className="right-container"
        >
          <button
            aria-label="Close Button"
            className="close-button"
            onClick={undefined}
          >
            <IconClose
              className="icon-close"
              data-name="Layer 1"
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
            />
          </button>
        </span>
      </span>
      <MenuDivider />
    </div>
  }
  onDisclose={[Function]}
  role="navigation"
/>
`;

exports[`MenuUtilityMenu should render a menu utility menu with onRequestClose set 1`] = `
<ContentContainer
  aria-label="Utility menu"
  className="utility-menu"
  fill={false}
  footer={null}
  header={
    <div
      className="header"
    >
      <span
        className="left-container"
      >
        <span
          className="inner-container"
        >
          <button
            aria-label="Back Button"
            className="back-button"
            onClick={[Function]}
          >
            <IconLeft
              className="icon-left"
              data-name="Layer 1"
              isBidi={true}
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
            />
          </button>
          <span
            className="header-text"
          >
            title
          </span>
        </span>
        <span
          className="right-container"
        >
          <button
            aria-label="Close Button"
            className="close-button"
            onClick={[Function]}
          >
            <IconClose
              className="icon-close"
              data-name="Layer 1"
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
            />
          </button>
        </span>
      </span>
      <MenuDivider />
    </div>
  }
  onDisclose={[Function]}
  role="navigation"
/>
`;
